tfresource/lib.w

```diff
@@ -0,0 +1,101 @@
+bring "cdktf" as cdktf;
+bring fs;
+bring cloud;
+bring util;
+
+pub class TerraformResource {
+  commands: MutJson;
+
+  new() {
+    this.setupProvider();
+
+    this.commands = {};
+
+    class Shell extends cdktf.TerraformResource {
+      c: MutJson;
+
+      new(commands: MutJson) {
+        super(terraformResourceType: "shell_script");
+        this.c = commands;
+      }
+
+      pub synthesizeAttributes(): Json {
+        return {
+          lifecycle_commands: Json.deepCopy(this.c),
+        };
+      }
+    }
+
+    new Shell(this.commands);
+  }
+
+  pub onCreate(handler: inflight (): Json) {
+    let prog = this.toExecutable("create", handler);
+    this.commands.set("create", "node {prog}");
+  }
+
+  pub onRead(handler: inflight (Json): Json) {
+    let prog = this.toExecutable("read", handler);
+    this.commands.set("read", "node {prog}");
+  }
+
+  pub onUpdate(handler: inflight (Json): Json) {
+    let prog = this.toExecutable("update", handler);
+    this.commands.set("update", "node {prog}");
+  }
+
+  pub onDelete(handler: inflight (Json): Json?) {
+    let prog = this.toExecutable("delete", handler);
+    this.commands.set("delete", "node {prog}");
+  }
+
+  toExecutable(name: str, handler: inflight (Json): Json?): str {
+    let workdir = std.Node.of(this).app.workdir;
+    let code: str = unsafeCast(handler)?._toInflight();
+
+    let path = fs.join(workdir, "{name}-{this.node.addr}.js");
+    let wrapper = "
```

Comment:
This wrapper is pretty cool - it basically packages the inflight handler as a node executable. Input is read as JSON from STDIN and return JSON is emitted to STDOUT.

Reply:
and the inflight code is bundled as any other inflight handler, right?

Reply:
@skorfmann what do you mean?

Reply:
it's just bundled with esbuild, so  I don't have to think about dependencies as a user

Reply:
since this is running from within the same project directory, dependencies are already installed in node_modules, so there's no need to bundle.

am I missing something?

Reply:
all good 👍